commit 539bbad457e7161f89fd4db3017b4abf478466f4
Author: Andrew Hsieh <andrewhsieh@google.com>
Date:   Mon Apr 14 16:55:29 2014 -0700

    [4.9] Patch GCC libstdc++ to work with Clang in C++11
    
    See 0e1e4c378ef579441c3815c6213422f1df530f9b
    
    Change-Id: I74314f368c202436ddef80639d7096e943d7fb5d

diff --git a/gcc-4.9/libstdc++-v3/include/std/type_traits b/gcc-4.9/libstdc++-v3/include/std/type_traits
index 4b434a6..0261c7d 100644
--- a/gcc-4.9/libstdc++-v3/include/std/type_traits
+++ b/gcc-4.9/libstdc++-v3/include/std/type_traits
@@ -1938,7 +1938,7 @@ _GLIBCXX_BEGIN_NAMESPACE_VERSION
     struct __common_type_impl
     : private __do_common_type_impl
     {
-      typedef decltype(_S_test<_Tp, _Up>(0)) type;
+      typedef typename decay<decltype(_S_test<_Tp, _Up>(0))>::type type;
     };
 
   struct __do_member_type_wrapper
